src/types/colorTable.cpp

```diff
@@ -561,13 +600,41 @@ try
     }
 
     std::string name(ss.str());
+    const auto standardVariantColorIter = xorgAppStandardVariantColorTable.find(name);
+    if (standardVariantColorIter != xorgAppStandardVariantColorTable.end())
+    {
+        const auto standardAndVariantBaseColor = standardVariantColorIter->second;
+        if (variantIndex > 4)
+        {
+            return std::nullopt;
+        }
+
+        if (foundVariant)
+        {
+            return _calculateXorgAppColorVariant(standardAndVariantBaseColor.at(1), variantIndex);
+        }
+        else
+        {
+            return standardAndVariantBaseColor.at(0);
+        }
+    }
+
     const auto variantColorIter = xorgAppVariantColorTable.find(name);
     if (variantColorIter != xorgAppVariantColorTable.end())
     {
-        const auto colors = variantColorIter->second;
-        if (variantIndex < colors.size())
+        const auto baseColor = variantColorIter->second;
+        if (variantIndex > 4)
+        {
+            return std::nullopt;
+        }
+
+        if (foundVariant)

```

Comment:
If `variantIndex` is 1 by default, this logic can be simplified. But I'm always hesitate to initialize some integer to 1 by default. It's just not as natural as 0..